common/src/main/java/io/netty/util/internal/PromiseNotificationUtil.java

```diff
@@ -66,9 +66,11 @@ public static void tryFailure(Promise<?> p, Throwable cause, InternalLogger logg
             if (err == null) {
                 logger.warn("Failed to mark a promise as failure because it has succeeded already: {}", p, cause);
             } else {
-                logger.warn(
+                if (logger.isWarnEnabled()) {
```

Comment:
Just merge this with the else and make it an else if?

Reply:
Got it